### PR TITLE
Client API for executing queries and awaiting results

### PR DESCRIPTION
**Pull request by jasonmitchell**
Reopening #1204 for issue #672

### CODE REVIEW

File: src/EventStore.ClientAPI/Projections/QueryManager.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            var delayInMilliseconds = initialDelayInMilliseconds * (Math.Pow(2, attempts) - 1);
+            delayInMilliseconds = Math.Min(delayInMilliseconds, maximumDelayInMilliseconds);
+
+            await Task.Delay(TimeSpan.FromMilliseconds(delayInMilliseconds));
+        }
+
+        private async Task<string> GetStatusAsync(string name, UserCredentials userCredentials)
+        {
+            var projectionStatus = await _projectionsManager.GetStatusAsync(name, userCredentials);
+            return projectionStatus.ParseJson<JObject>()["status"].ToString();
```
Comment (gregoryyoung) [inline src/EventStore.ClientAPI/Projections/QueryManager.cs]: We probably want these passed in as timespans

Comment (jasonmitchell) [inline src/EventStore.ClientAPI/Projections/QueryManager.cs]: Makes sense, I'll make this change.

**Comment by pgermishuys** [comment]
Hi @jasonmitchell, it would seem that a rebase has gone wrong here. We had to repurpose the 4.0.1 release last week and this would likely need to go into 4.0.2. Sorry once again.

**Comment by jasonmitchell** [comment]
No problem, I'll rebase onto 4.0.2 when I'm home from work and update the PR.

**Pull request merged**